pool.py

```diff
@@ -0,0 +1,71 @@
+from dice import Dice
+
+
+class ContainerEmptyError(Exception):
+    """
+    Raised whenever an operation is attempted on an empty container object
+    which should only be attempted when the container is not empty
+    """
+    pass
+
+
+class DicePool:
+    """
+    A pool of dice. That can be rolled.
+    """
+
+    def __init__(self, dice=[]):
+        """
+        Instantiates the dice pool object.
+
+        dice: A list of dice in the pool.
+        """
+
+        if type(dice) is not list:
+            raise TypeError(
+                'dice argument {} to DicePool object should be a list'
+                .format(str(dice))
+            )
+
+        for d in dice:
+            if not isinstance(d, Dice):
+                raise TypeError(
+                    'Element {} of dice argument to DicePool object should be a Dice'
+                    .format(str(d)))
+
+        self.dice = dice
+
+    def __repr__(self):
+        return repr(self.dice)
+
+    def __str__(self):
+        return str(self.dice)
+
+    def add_dice(self, dice):
+        """Adds the dice to the pool. dice must be a Dice object"""
+
+        if not isinstance(dice, Dice):
```

Comment:
Re my earlier comment about types, this isn't prey to the subclass issue, but it does still assume that only subclasses of `Dice` can be used. Maybe that's the case, but maybe what we actually care about is whether they have a `roll()` method?